### PR TITLE
Jest: Add setSystemTime and getRealSystemTime fn types

### PR DESCRIPTION
**Pull request by liamfd**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
  - https://jestjs.io/docs/en/jest-object#jestsetsystemtime, https://jestjs.io/docs/en/jest-object#jestgetrealsystemtime
  - These are based on Jest's internal types, but I've got a PR open to update them, this PR assumes that update has happened on their end: https://github.com/facebook/jest/pull/10169 (you can see the internal types I'm using in that PR).
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
  - Not sure if I'll need to make a change here - these types are valid and work at runtime for the current version of jest. The PR I have open for essentially just updates jest's internal types and docs to reflect that, so I'm not sure if it would be better to update the version reference, given that it should be compatible with both. Regardless, the current header says `26.0` and if my jest PR is accepted I would assume it would be a patch release.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/jest/index.d.ts
Comment (liamfd) [inline types/jest/index.d.ts:144]: I tried to follow the patterns I observed elsewhere in this file here, I just copy pasted these comments straight out of jest's docs.

File: types/jest/index.d.ts
Comment (liamfd) [inline types/jest/index.d.ts:143]: I tried to follow the comment patterns I observed elsewhere in this file, I just copy pasted this comment and the one below straight out of jest's docs.

File: types/jest/index.d.ts
Comment (liamfd) [inline types/jest/index.d.ts:144]: As mentioned in the PR description, jest currently types this as:

```ts
    function setSystemTime(now?: number): void;
```

I have [a PR open](https://github.com/facebook/jest/pull/10169) to expand it to support `Date`, given that under the hood `setSystemTime` is calling a function that accepts both. If they do not want to support that signature, I will remove it here.

Comment (wsmd) [reply types/jest/index.d.ts:144]: Perhaps we can hold off on adding support for `Date` until the other PR is merged to keep things consistent and work as intended? Once that other PR is merged and a newer version of Jest is released with the `Date` support, we can then update the types here to reflect that latest changes?

**Review by wsmd — CHANGES_REQUESTED** [review]
Thanks for the PR @liamfd! This looks good overall. I left one comment suggesting that we hold off on adding support for `Date` for now.

File: types/jest/index.d.ts
Comment (liamfd) [reply types/jest/index.d.ts]: Sounds good to me

**Review by wsmd — APPROVED** [review]
Thanks @liamfd!

**Pull request merged**
